### PR TITLE
Added uri to path when using file protocol to chdir

### PR DESCRIPTION
**Pull request by jkutner**
See jruby/warbler#323

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyDir.java [hunk 1]
```diff
@@ -326,6 +326,9 @@ public static IRubyObject chdir(ThreadContext context, IRubyObject recv, IRubyOb
         if (adjustedPath.startsWith("uri:")){
             realPath = adjustedPath;
         }
+        else if (adjustedPath.startsWith("file:")){
+            realPath = "uri:" + adjustedPath;
+        }
```
Comment (mkristian) [inline core/src/main/java/org/jruby/RubyDir.java]: a path with ```!/``` is treated as jar uri whether this is a file: or jar: or jar:file: prefix or not. and this is the case where you want the uri: to add. but for ```uri:``` to work properly it needs ```new URL(adjustedPath).openStream()``` to work. i.e. in your case make sure there is jar:file: prefix

an path with file: onto the filesystem without the ```!/``` better just remove the file: prefix and let jruby use the filesystem.

but I do not understand why your current patch fixes the warbler problem

Comment (mkristian) [inline core/src/main/java/org/jruby/RubyDir.java]: I think we all to chdir to all paths following this pattern: https://github.com/jruby/jruby/blob/jruby-1_7/core/src/main/java/org/jruby/RubyFile.java#L670

it feels wrong to switch one pseudo-uri to another

**Comment by jkutner** [comment]
No, neither 1.7.21 nor 1.7.22.dev fixed this. I tried removing the `file:` prefix, and got this error:

```
java.net.MalformedURLException: no protocol: /Users/jkutner/workspace/warbler-debug/warbler-debug.war!/WEB-INF/gems/bundler/gems/ruby-units-a777aeebcfbd/lib/ruby_units/version.rb
	at java.net.URL.<init>(URL.java:586)
	at java.net.URL.<init>(URL.java:483)
	at java.net.URL.<init>(URL.java:432)
	at org.jruby.util.URLResource.createRegularURI(URLResource.java:199)
	at org.jruby.util.URLResource.create(URLResource.java:187)
	at org.jruby.util.JRubyFile.createResource(JRubyFile.java:91)
	at org.jruby.util.JRubyFile.createResource(JRubyFile.java:102)
	at org.jruby.util.JRubyFile.createResource(JRubyFile.java:75)
	at org.jruby.RubyFileStat.setup(RubyFileStat.java:122)
	at org.jruby.RubyFileStat.newFileStat(RubyFileStat.java:97)
	at org.jruby.Ruby.newFileStat(Ruby.java:3342)
	at org.jruby.RubyFile.stat(RubyFile.java:894)
```

I will investigate the .jrubydir

**Comment by mkristian** [comment]
the .jrubydir will not help here with this concrete bundler problem. just passing on the ```realPath = adjustedPath;``` should help and keeps things more consistent.

yes, I did not run the sample repo at the beginning.

**Comment by jkutner** [comment]
Ah, i understand now. I will improve this

**Comment by mkristian** [comment]
never saw the update - but saw your question on IRC :)

**Comment by jkutner** [comment]
@mkristian thanks for your help with this!

**Pull request merged**